scheduler/job.c

```diff
@@ -3123,8 +3124,9 @@ finalize_job(cupsd_job_t *job,		/* I - Job */
   * Similarly, clear the "offline-report" reason for non-USB devices since we
   * rarely have current information for network devices...
   */
+  sscanf(job->printer->device_uri, "%254[^:]", scheme);
 
-  if (!strstr(job->printer->device_uri, "usb:"))
+  if (!strstr(scheme, "usb"))
```

Comment:
This should just be `strcmp(scheme, "usb")`.

Reply:
Really, what about the ippusb device? scheduler/ipp.c Line 2711

Reply:
Hmm, yeah. Reading up on commit fc01abc173 it sounds like what we need here is a suffix match, so `strlen(scheme) < 3 || strcmp(scheme + strlen(scheme) - 3, "usb")` will do the trick. Just looking for "usb" in the scheme isn't right because it will match anywhere in the string, whereas before we included the scheme delimiter in the search.

Alternately you can keep the original code and only use "scheme" when reporting an error.